### PR TITLE
#571 Default parameter for text input

### PR DESCRIPTION
**Pull request by CarlosNZ**
Applies a "default" parameter to shortText and longText plugins, which allows a certain value to be "pre-filled" when launching an application. 

This will be useful for an "Edit user details" form, where the fields can be pre-filled with existing values.

Have made a back-end to demonstrate this,  but it's just a rudimentary "Edit User" template, without full functionality.

Use branch from [back-end PR #332](https://github.com/openmsupply/application-manager-server/pull/332)

To see the "default" values in use, start a UserEdit template with: http://localhost:3000/application/new?type=UserEdit

### CODE REVIEW

File: src/formElementPlugins/longText/src/ApplicationView.tsx [hunk 1]
```diff
@@ -14,12 +14,15 @@ const ApplicationView: React.FC<ApplicationViewProps> = ({
...
-  const { label, description, placeholder, lines, maxLength } = parameters
+  const { label, description, placeholder, lines, default: defaultValue, maxLength } = parameters
 
   const { isEditable } = element
 
   useEffect(() => {
-    onUpdate(value)
+    if (!value && defaultValue) {
+      onSave({ text: defaultValue })
+      setValue(defaultValue)
```
Comment (andreievg) [inline src/formElementPlugins/longText/src/ApplicationView.tsx:24]: I am getting confused now, how the ApplicationViewWrapper interacts with ApplciationView and useGetFullApplicationStructure, sound like we need another diagram : /

I am not sure what the difference here is between calling onUpdate and not calling it

**Review by andreievg — APPROVED** [review]
Approved, but need to compile a digram sometime (and in depth one), I'll try to do it towards the end of the week (if you haven't done it already)

**Comment by CarlosNZ** [comment]
> Approved, but need to compile a digram sometime (and in depth one), I'll try to do it towards the end of the week (if you haven't done it already)

Okay, have made an issue, hopefully get to it soon (ish) :)

File: src/formElementPlugins/longText/src/ApplicationView.tsx (hunk 1 above)
Comment (CarlosNZ) [reply src/formElementPlugins/longText/src/ApplicationView.tsx:24]: onUpdate just updates the local state "value", whereas onSave saves it to the database. However, onSave *includes* a call to onUpdate within it, so no need to do onUpdate if you're doing onSave :)

**Pull request merged**